packages/remix-react/browser.tsx

```diff
@@ -381,6 +425,88 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
     });
   }, [location]);
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  let fogOfWarAbortControllerRef = React.useRef<AbortController | undefined>();
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  React.useEffect(() => {
+    // Don't perform prefetching without flag or if the user has `saveData` enabled
+    if (!fogOfWar || navigator.connection?.saveData === true) {
+      return;
+    }
+
+    async function fetchPatches() {
+      fogOfWarAbortControllerRef.current?.abort();
+
+      let lazyPaths = getFogOfWarPaths(
+        fogOfWar!,
+        window.__remixContext.basename
+      );
+      if (lazyPaths.length === 0) {
+        return;
+      }
```

Comment:
Short circuit there are no actively rendered paths to fetch - which means that they are all either matchable via the currently known routes, or they are a known 404 from a prior fetch.